From b014befbd70d3a0c53733e0e74bbbb2e31d55ecf Mon Sep 17 00:00:00 2001
From: zhoupeng <zhoupeng@loongson.cn>
Date: Mon, 16 Nov 2020 19:26:53 +0800
Subject: [PATCH] Modify the rendering driver score.

1.Modified the following drivers:
 -vdpau  0   -> 300
 -xv     200 -> 290
 -x11    100 -> 280
2.Scores of commonly used rendering drivers on mips platforms:
 vdpau(300)/xv(290)/x11(280)/opengl(270)/opengl-es2(265)/wayland(170)

Change-Id: I7b1ff0205e0044c0ed1cdf59c435211096e78293

---
 modules/hw/vdpau/display.c        | 4 ++++
 modules/video_output/xcb/x11.c    | 4 ++++
 modules/video_output/xcb/xvideo.c | 4 ++++
 3 files changed, 12 insertions(+)

diff --git a/modules/hw/vdpau/display.c b/modules/hw/vdpau/display.c
index edd6096729..c5da42c695 100644
--- a/modules/hw/vdpau/display.c
+++ b/modules/hw/vdpau/display.c
@@ -46,7 +46,11 @@ vlc_module_begin()
     set_description(N_("VDPAU output"))
     set_category(CAT_VIDEO)
     set_subcategory(SUBCAT_VIDEO_VOUT)
+#ifdef _mips
+    set_capability("vout display", 300)
+#else
     set_capability("vout display", 0)
+#endif
     set_callbacks(Open, Close)
 
     add_shortcut("vdpau")
diff --git a/modules/video_output/xcb/x11.c b/modules/video_output/xcb/x11.c
index ef671c84f5..2c7d2a2a6f 100644
--- a/modules/video_output/xcb/x11.c
+++ b/modules/video_output/xcb/x11.c
@@ -49,7 +49,11 @@ vlc_module_begin ()
     set_description (N_("X11 video output (XCB)"))
     set_category (CAT_VIDEO)
     set_subcategory (SUBCAT_VIDEO_VOUT)
+#ifdef _mips
+    set_capability ("vout display", 280)
+#else
     set_capability ("vout display", 100)
+#endif
     set_callbacks (Open, Close)
     add_shortcut ("xcb-x11", "x11")
 
diff --git a/modules/video_output/xcb/xvideo.c b/modules/video_output/xcb/xvideo.c
index e79b013b69..7bed49e7e4 100644
--- a/modules/video_output/xcb/xvideo.c
+++ b/modules/video_output/xcb/xvideo.c
@@ -63,7 +63,11 @@ vlc_module_begin ()
     set_description (N_("XVideo output (XCB)"))
     set_category (CAT_VIDEO)
     set_subcategory (SUBCAT_VIDEO_VOUT)
+#ifdef _mips
+    set_capability ("vout display", 290)
+#else
     set_capability ("vout display", 200)
+#endif
     set_callbacks (Open, Close)
 
     add_integer ("xvideo-adaptor", -1,
-- 
2.20.1

